src/Security/Member.php

```diff
@@ -1702,4 +1710,46 @@ public function getHtmlEditorConfigForCMS()
         // If can't find a suitable editor, just default to cms
         return $currentName ? $currentName : 'cms';
     }
+
+    /**
+     * This needs to be public because it's referenced in getMemberPasswordField() and called in ConfirmedPasswordField
+     */
+    public static function generateRandomPassword(array $args = []): string
+    {
+        $password = '';
+        $validator = self::password_validator();
+        $minLength = $validator ? $validator->getMinLength() : 20;
+        $maxLength = $args['maxLength'] ?? 0;
+        if ($maxLength === 0) {
+            $maxLength = max($minLength, 20);
+        } else {
+            if ($maxLength < $minLength) {
+                throw new InvalidArgumentException('maxLength argument is less than PasswordValidator minLength');
+            }
+        }
+        $dummyMember = new Member();
+        // These are similar character sets that used in the default PasswordValidator
```

Comment:
I don't understand what this comment means, can you please explain?
If it means these character sets are already stored in `PasswordValidator`, why aren't we reusing them instead of recreating them?

Reply:
Still not done?

Reply:
Updated